src/Provider/Patreon.php

```diff
@@ -45,10 +45,12 @@ protected function initialize()
     {
         parent::initialize();
 
-        $this->tokenRefreshParameters += [
-            'client_id' => $this->clientId,
-            'client_secret' => $this->clientSecret,
-        ];
+        if (is_array($this->tokenRefreshParameters)) {
```

Comment:
I've feeling we can replace `is_array($this->tokenRefreshParameters)` to `$this->isRefreshTokenAvailable()` everywhere